dynamodb/dynamodb.sim.w

```diff
@@ -0,0 +1,214 @@
+bring sim;
+bring util;
+bring cloud;
+bring "./dynamodb-types.w" as dynamodb_types;
+bring "./dynamodb-client.w" as dynamodb_client;
+
+interface Process {
+  inflight kill(): void;
+}
+
+struct SpawnOptions {
+  command: str;
+  arguments: Array<str>;
+  onData: inflight (str): void;
+}
+
+interface Client {
+  inflight createTable(input: Json): Json;
+  inflight deleteTable(input: Json): Json;
+  inflight updateTimeToLive(input: Json): Json;
+}
+
+struct CreateClientOptions {
+  endpoint: str;
+  region: str;
+  credentials: Json;
+}
+
+class Util {
+  extern "./dynamodb.mjs" pub static inflight getPort(): num;
+  extern "./dynamodb.mjs" pub static inflight spawn(options: SpawnOptions): Process;
+  extern "./dynamodb.mjs" pub static inflight createClient(options: CreateClientOptions): Client;
+  extern "./dynamodb.mjs" pub static inflight processRecordsAsync(endpoint: str, tableName: str, handler: inflight (dynamodb_types.StreamRecord): void): void;
+}
+
+class Host {
+  pub endpoint: str;
+
+  new() {
+    let containerName = "winglibs-dynamodb-{util.uuidv4()}";
+
+    let state = new sim.State();
+    this.endpoint = state.token("endpoint");
+
+    new cloud.Service(inflight () => {
+      let port = Util.getPort();
+
+      let docker = Util.spawn(
```

Comment:
Would it make sense to use containers.Workload here?

Reply:
I guess, I just tried it and the resulting code is much cleaner, but I noticed that the containers doesn't get removed when I Ctrl+C the `wing it` process.

Maybe we can improve the containers winglib. The difference is that I spawn the docker container without detach and with the `--rm` option.